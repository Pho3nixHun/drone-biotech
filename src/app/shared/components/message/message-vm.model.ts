export interface MessageVM {
  title: string;
  descriptionKey: string;
}
