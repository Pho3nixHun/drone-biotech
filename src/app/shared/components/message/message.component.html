<div class="bg-gray-700 p-6 rounded-xl">
    <h1
        class="text-6xl sm:text-8xl font-bold text-white mb-4 text-center w-full"
    >
        {{ vm().title }}
    </h1>
    <p class="text-white md:text-xl max-w-md">
        {{ vm().descriptionKey | transloco }}
    </p>
</div>
