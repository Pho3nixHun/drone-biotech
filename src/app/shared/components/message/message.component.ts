import { Component, input } from '@angular/core';
import { MessageVM } from './message-vm.model';
import { TranslocoModule } from '@jsverse/transloco';

/**
 * MessageComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering a message.
 * - Manages the layout and styling of a message.
 *
 * Out-of-Scope:
 * - Should not handle the data fetching or business logic related to message.
 * - Should not be responsible for manipulating or transforming data.
 */

@Component({
  selector: 'app-message',
  standalone: true,
  imports: [TranslocoModule],
  templateUrl: './message.component.html',
})
export class MessageComponent {
  public vm = input.required<MessageVM>();
}
