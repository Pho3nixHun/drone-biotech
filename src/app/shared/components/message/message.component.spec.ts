import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MessageComponent } from './message.component';
import { MessageVM } from './message-vm.model';
import { getTranslocoModule } from 'transloco-testing.module';

const en = { description: 'description' };
describe('MessageComponent', () => {
    let fixture: ComponentFixture<MessageComponent>;
    let compiled: HTMLElement;

    const vm: MessageVM = {
        title: '404!',
        descriptionKey: en.description,
    };

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [
                MessageComponent,
                getTranslocoModule({
                    langs: { en },
                    translocoConfig: {
                        availableLangs: ['en'],
                        defaultLang: 'en',
                    },
                }),
            ],
        }).compileComponents();

        fixture = TestBed.createComponent(MessageComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    //Snapshot test
    it('should render the template when there is a VM provided', () => {
        //Arrange
        fixture.componentRef.setInput('vm', vm);
        //Act
        fixture.detectChanges();
        //Assert
        expect(compiled).toMatchSnapshot();
    });
});
