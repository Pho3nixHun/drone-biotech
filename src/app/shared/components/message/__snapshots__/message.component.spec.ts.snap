// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageComponent should render the template when there is a VM provided 1`] = `
<div
  id="root0"
  ng-version="18.2.8"
>
  <div
    class="bg-gray-700 p-6 rounded-xl"
  >
    <h1
      class="text-6xl sm:text-8xl font-bold text-white mb-4 text-center w-full"
    >
       404! 
    </h1>
    <p
      class="text-white md:text-xl max-w-md"
    >
       description 
    </p>
  </div>
</div>
`;
